c/extras/aggregator.cc

```diff
@@ -385,12 +390,15 @@ void Aggregator::project_row(DataTablePtr& dt_exemplars, double* r, int32_t row_
   std::memset(r, 0, static_cast<size_t>(max_dimensions) * sizeof(double));
 
   for (int32_t i = 0; i < (dt_exemplars->ncols) * max_dimensions; ++i) {
```

Comment:
Looks like what you really wanted to have here is a double-loop with `col_id` running from 0 to `dt_exemplars->ncols`, and `row_id` from 0 to `max_dimensions`. Then you won't have to recompute `c`, `d_c`, etc on each iteration.

Reply:
At some point I had a double loop here for the `double**` matrix and then replaced it with the `double*` one. Yes, obviously need to take another look at its implementation. Thanks!

Reply:
You can still have a single-dimensional matrix, indexed by `irow * nrows +  icol` or whatever

Reply:
Yes, sure, that is what I missed at the time of re-implementation...

Reply:
I guess it had something to do with the matrix by vector multiplication I do here. I will double check it. Thanks.